### PR TITLE
[mysql-persistence] Do not log plain passwords

### PR DESCRIPTION
**Pull request by martinvw**
Fixes: https://github.com/openhab/openhab-distro/issues/762

Signed-off-by: Martin van Wingerden <martin@martinvw.nl>

### CODE REVIEW

File: bundles/persistence/org.openhab.persistence.mysql/java/org/openhab/persistence/mysql/internal/MysqlPersistenceService.java [hunk 1]
```diff
@@ -553,7 +553,7 @@ private void connectToDatabase() {
         } catch (Exception e) {
             logger.error(
                     "mySQL: Failed connecting to the SQL database using: driverClass={}, url={}, user={}, password={}",
-                    driverClass, url, user, password, e);
+                    driverClass, url, user, password.replaceAll(".", "*"), e);
```
Comment (kaikreuzer) [inline bundles/persistence/org.openhab.persistence.mysql/java/org/openhab/persistence/mysql/internal/MysqlPersistenceService.java]: You are still exposing some information about the password here (its length), which should also be avoided, see the comment at https://github.com/eclipse/smarthome/issues/4583#issuecomment-349715045.

So I'd suggest to either put 8 asterisks or maybe simply remove the password part completely here as it anyhow is not helpful.

Comment (martinvw) [reply bundles/persistence/org.openhab.persistence.mysql/java/org/openhab/persistence/mysql/internal/MysqlPersistenceService.java]: @kaikreuzer, I like that, shall I also remove it then in all other places in this repo.

Should I also apply it for trace logging? or is that configurable and thus okay.

Comment (kaikreuzer) [reply bundles/persistence/org.openhab.persistence.mysql/java/org/openhab/persistence/mysql/internal/MysqlPersistenceService.java]: Yes, can be removed from other places as well from the logging.
I'd also remove it from trace statements.

**Comment by martinvw** [comment]
Hi @kaikreuzer, its updated!

**Comment by CrispinP** [comment]
Thank you

**Comment by martinvw** [comment]
@CrispinP thank you for reporting it. 

Note that I also created follow up issue for openHAB2 addons

https://github.com/openhab/openhab2-addons/issues/3990

**Pull request merged**